camviewer_ui_impl.py

```diff
@@ -1297,6 +1307,44 @@ def onSizeUpdate(self):
         except Exception:
             pass
 
+    def user_set_max_image_rate(self, rate: int) -> None:
+        """
+        Call set_max_image_rate and record the value as the desired rate.
+
+        This desired rate is used so that after a timeout, we can
+        restore the GUI to the user's last desired rate when they
+        interact with the GUI again.
+        """
+        self.last_des_max_rate = rate
+        self.set_max_image_rate(rate)
+
+    def set_max_image_rate(self, rate: int) -> None:
+        """
+        Update the maximum allowed rate by restarting the appropriate timer.
+
+        Rate is expected to be a positive integer in Hz.
+
+        When this timer expires, wantImage is called.
+        If a new image is available at this time, it will be fetched and rendered.
+
+        The image rate will determine the rate limiting timeout.
+        1 Hz will never time out
+        5 Hz or less will time out in a week
+        30 Hz and greater will time out in a day
+        """
+        rate = int(rate)
+        if rate <= 0:
+            raise ValueError("Rate must be greater than zero!")
+        self.imageTimer.start(int(1000.0 / rate))
+        self.ui.label_max_rate_value.setText(f"{rate} Hz")
```

Comment:
When I check this out my ui file doesn't have this attribute

<details><summary>Details</summary>
<p>

```python
Traceback (most recent call last):
  File "/cds/home/r/roberttk/src/camviewer/./camviewer.pyw", line 94, in <module>
    gui = GraphicUserInterface(
  File "/cds/home/r/roberttk/src/camviewer/camviewer_ui_impl.py", line 511, in __init__
    self.user_set_max_image_rate(rate)
  File "/cds/home/r/roberttk/src/camviewer/camviewer_ui_impl.py", line 1319, in user_set_max_image_rate
    self.set_max_image_rate(rate)
  File "/cds/home/r/roberttk/src/camviewer/camviewer_ui_impl.py", line 1339, in set_max_image_rate
    self.ui.label_max_rate_value.setText(f"{rate} Hz")
AttributeError: 'Ui_MainWindow' object has no attribute 'label_max_rate_value'
```

</p>
</details>

Reply:
`camviewer` doesn't use the ui file at runtime, it expects you to `make` an equivalent `.py` file

Reply:
oh no I forgot to make 😮‍💨